hist/hist/src/THnSparse.cxx

```diff
@@ -942,6 +942,31 @@ void THnSparse::AddBinError2(Long64_t bin, Double_t e2)
    (*chunk->fSumw2)[bin % fChunkSize] += e2;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Scale contents and errors of this histogram by c:
+/// this = this * c
+/// It does not modify the histogram's number of entries.
+
+void THnSparse::Scale(Double_t c)
+{
+
+   Double_t nEntries = GetEntries();
+   // Scale the contents & errors
+   Bool_t haveErrors = GetCalculateErrors();
+   Long64_t i = 0;
+   THnSparseBinIter iter(kFALSE, this);
```

Comment:
```suggestion
   std::unique_ptr<ROOT::Internal::THnBaseBinIter> iter{CreateIter(false)};
```
There is already a virtual method to create the right iterator. Can we just use that one in `THnBase::Scale()`? Then you don't need to make `Scale()` virtual.

Also, there are many occurrences on explicit `THnIter` instantiations in `THnBase`, e.g. also in `Multiply()`. Can we replace all of them with the virtual iterator creation dispatch in one go?

Reply:
Done! thanks for the suggestion